.github/workflows/build.yml

```diff
@@ -311,16 +311,99 @@ jobs:
       - name: Build wheels
         uses: pypa/cibuildwheel@v2.17.0
         with:
           package-dir: Core/ClientSMLSWIG/Python/
 
+      - name: ABI3Audit (*nix)
+        if: matrix.os != 'windows-latest'
+        run: pipx run abi3audit -S -v wheelhouse/*
+
+      - name: ABI3Audit (Windows)
+        if: matrix.os == 'windows-latest'
+        # For windows we can't use glob expansion, so we need to enumerate the files manually.
+        # https://stackoverflow.com/a/16804630/8700553
+        run: Get-ChildItem -File wheelhouse | Foreach {pipx run abi3audit -S -v $_.fullname}
+
       - uses: actions/upload-artifact@v4
         with:
           name: cibw-wheels-${{ matrix.os }}-${{ strategy.job-index }}
           path: ./wheelhouse/*.whl
 
+  test_python:
+    name: Test wheel on all python versions
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          # linux target which supports all python versions we want to install for
+          - ubuntu-20.04
+            # latest available X86_64 target
+          - macos-12
+            # latest is ARM
+          - macos-latest
+        python-minor:
+          - 5
+          - 6
+          - 7
+          - 8
+          - 9
+          - 10
+          - 11
+          - 12
+
+        # for macos-latest, we can't use python 3.5 - 3.7
+        exclude:
+          - os: macos-latest
+            python-minor: 5
+          - os: macos-latest
+            python-minor: 6
+          - os: macos-latest
+            python-minor: 7
+
+    needs:
+      # We depend on the python wheel builds because we need their artifacts
+      - python_wheels
+
+      # We depend on the builds themselves to gatekeep the upload if build + testing fails,
+      # usually then also the python wheel build should fail, but the normal builds do more thorough checking.
+      - Posix
+      - Windows
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: cibw-wheels-*
+          path: wheelhouse/
+          merge-multiple: true
+
+      - name: Setup Base Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: |
+            3.${{ matrix.python-minor }}
+            3.12
+          # Latest version is installed under `python`
+        env:
+          # Older python versions fail install if we dont do this: https://github.com/actions/setup-python/issues/866
+          PIP_TRUSTED_HOST: "pypi.python.org pypi.org files.pythonhosted.org"
+
+      - name: Run python test command
+        # TODO: replace with actual test command: https://github.com/SoarGroup/Soar/issues/460
+        run: |
+          pip3.${{ matrix.python-minor }} install soar-sml -f wheelhouse --no-index
+          python3.${{ matrix.python-minor }} -c "import soar_sml;
+          k=soar_sml.Kernel.CreateKernelInNewThread();
+          a=k.CreateAgent('soar');
+          assert(a.ExecuteCommandLine('echo hello world').strip()=='hello world')"
```

Comment:
Currently i've just taken this from the cibuildwheel command, but do you want me to make an effort to address #460 in this PR as well?

Reply:
It's up to you. I wouldn't make it a requirement of finishing this PR.

Reply:
The risk is that in a follow-up PR of adding this, we may encounter real test failures that I wouldn't be able to easily resolve, and/or would mean that some versions are effectively unsupported.

If that's alright, at least for the development phase of .3, I'd say this is alright to merge if you say so :)

Reply:
Okay, you actually make a very good point. Since we're moving to a single abi3 library for each platform, if that breaks with any particular Python version then this will be a pain to fix, or we'd actually just need to roll it back or something. I think we should take care of #460 .

Reply:
Done!